x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -324,6 +327,35 @@ private LogicalPlan parse(String query, QueryParams params) {
         return parsed;
     }
 
+    static void handleFieldCapsFailures(
```

Comment:
This probably could use some javadoc to describe the logic of what's happening inside.

Reply:
added in https://github.com/elastic/elasticsearch/pull/130840/commits/9b19452a4f4c19e3f29b6ae668d965a94a929909